sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -1152,9 +1152,11 @@ func withDynamicResolver(ctx context.Context, ref string, registryConfig *Regist
 		}
 	// For Amazon ECR Public registries, we should try and fetch credentials before resolving the image reference
 	case strings.HasPrefix(ref, "public.ecr.aws/"):
-		// ... not if the user has specified their own registry credentials for 'public.ecr.aws'; In that case we use the default resolver.
-		if _, found := registryConfig.Credentials["public.ecr.aws"]; found {
-			return defaultResolver
+		if registryConfig != nil {
```

Comment:
Isn't the right behavior here to just return the defaultResolver if `registryConfig` is nil?

Reply:
Perhaps a comment is in order. Assuming that my version of "what this thing does" is correct.